### PR TITLE
Use extracted guestcluster library 

### PR DESCRIPTION
**Pull request by rossf7**
The library got extracted to https://github.com/giantswarm/guestcluster and this updates cluster-operator to use it and deletes the old code.

### CODE REVIEW

File: pkg/v6/guestcluster/guestcluster.go [hunk 1]
```diff
@@ -1,145 +0,0 @@
-package guestcluster
```
Comment (rossf7) [inline pkg/v6/guestcluster/guestcluster.go]: Old service gets removed from pkg/v6

File: pkg/v6/resource/chart/create.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 	}
 
 	guestHelmClient, err := r.getGuestHelmClient(ctx, obj)
-	if guestcluster.IsNotFound(err) {
+	if guestcluster.IsTimeout(err) {
```
Comment (rossf7) [inline pkg/v6/resource/chart/create.go]: Interface changed as part of extracting to the library.

File: service/controller/azure/v6/resource_set.go [hunk 3]
```diff
@@ -130,6 +130,8 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 		c := guestcluster.Config{
 			CertsSearcher: config.CertSearcher,
 			Logger:        config.Logger,
+
+			CertID: certs.ClusterOperatorAPICert,
```
Comment (rossf7) [inline service/controller/azure/v6/resource_set.go]: Another change that was made in the library. Ensures we search for the correct cert.

File: pkg/v6/resource/chartconfig/resource.go [hunk 4]
```diff
@@ -11,8 +11,8 @@ import (
 	"k8s.io/client-go/kubernetes"
 
 	"github.com/giantswarm/cluster-operator/pkg/cluster"
-	"github.com/giantswarm/cluster-operator/pkg/v6/guestcluster"
 	"github.com/giantswarm/cluster-operator/pkg/v6/key"
+	"github.com/giantswarm/guestcluster"
```
Comment (xh3b4sd) [inline pkg/v6/resource/chartconfig/resource.go]: This should be moved. There are a couple of occasions of this.

Comment (rossf7) [reply pkg/v6/resource/chartconfig/resource.go]: Good catch, thanks. Fixed.

**Pull request merged**